front/admin/cli.ts

```diff
@@ -0,0 +1,345 @@
+import { planForWorkspace } from "@app/lib/auth";
+import { Membership, User, Workspace } from "@app/lib/models";
+import { new_id } from "@app/lib/utils";
+import parseArgs from "minimist";
+
+// `cli` takes an object type and a command as first two arguments and then a list of arguments.
+
+const workspace = async (command: string, args: parseArgs.ParsedArgs) => {
+  switch (command) {
+    case "find": {
+      if (!args.name) {
+        throw new Error("Missing --name argument");
+      }
+
+      const workspaces = await Workspace.findAll({
+        where: {
+          name: args.name,
+        },
+      });
+
+      workspaces.forEach((w) => {
+        console.log(
+          `> wId='${w.sId}' name='${w.name}' description="${w.description}"`
+        );
+      });
+      return;
+    }
+    case "show": {
+      if (!args.wId) {
+        throw new Error("Missing --wId argument");
+      }
+
+      const w = await Workspace.findOne({
+        where: {
+          sId: args.wId,
+        },
+      });
+
+      if (!w) {
+        throw new Error(`Workspace not found: wId='${args.wId}'`);
+      }
+
+      console.log(`workspace:`);
```

Comment:
You could define a little interface and pretty print it if it makes your life easier:

```
interface adultWithKids {
  name:  string;
  age: number;
  kids: {
    name: string;
    age: number;
    sex: string;
  }[]
}

const arico: adultWithKids = {
  name:"aric lasry",
  age: 32,
  kids: [
    {
      name: "milo lasry",
      age: 2,
      sex:"M" 
    },
    {
      name: "luna lasry",
      age: 2,
      sex:"F"
    }
  ]
}

console.log(JSON.stringify(arico, null, 2))
```

Reply:
```
{
  "name": "aric lasry",
  "age": 32,
  "kids": [
    {
      "name": "milo lasry",
      "age": 2,
      "sex": "M"
    },
    {
      "name": "luna lasry",
      "age": 2,
      "sex": "F"
    }
  ]
}
```